File: WebLeave/API/Models/LunchBreak.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace API.Models
{
    public partial class LunchBreak
    {
        [Key]
        public int Id { get; set; }
        [Required]
        [StringLength(20)]
        public string Key { get; set; }
        [Precision(0)]
        public TimeSpan WorkTimeStart { get; set; }
        [Precision(0)]
        public TimeSpan WorkTimeEnd { get; set; }
        [Precision(0)]
        public TimeSpan LunchTimeStart { get; set; }
        [Precision(0)]
        public TimeSpan LunchTimeEnd { get; set; }
        [StringLength(250)]
        public string Value_vi { get; set; }
        [StringLength(250)]
        public string Value_en { get; set; }
        [StringLength(250)]
        public string Value_zh { get; set; }
        public int? Seq { get; set; }
        public bool? Visible { get; set; }
        public int? CreatedBy { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? CreatedTime { get; set; }
        public int? UpdatedBy { get; set; }
        [Column(TypeName = "datetime")]
        public DateTime? UpdatedTime { get; set; }
    }
}

File: WebLeave/SPA/src/app/_core/models/common/lunch-break.ts
export interface LunchBreak {
    id: number;
    key: string;
    workTimeStart: string;
    workTimeEnd: string;
    lunchTimeStart: string;
    lunchTimeEnd: string;
    value_vi: string;
    value_en: string;
    value_zh: string;
    seq: number | null;
    visible: boolean | null;
    createdBy: number | null;
    createdTime: string | null;
    updatedBy: number | null;
    updatedTime: string | null;
    type?: string;
}

export interface WorkShift {
    key: string;
    workTimeStart: Time;
    workTimeEnd: Time;
    lunchTimeStart: Time;
    lunchTimeEnd: Time;
}

export interface Time {
    hours: number;
    minutes: number;
    seconds: number;
}